src/sentry/incidents/endpoints/project_alert_rule_index.py

```diff
@@ -1,17 +1,132 @@
+                "type": "alert-rule",
+                "id": six.text_type(rule.id),
+                "name": rule.name,
+                "organizationId": six.text_type(rule.organization_id),
+                "status": rule.status,
+                # TODO: Remove when frontend isn't using
+                "thresholdType": 0,
+                "dataset": rule.dataset,
+                "query": rule.query,
+                "aggregation": rule.aggregation,
+                "aggregations": [rule.aggregation],
+                "timeWindow": rule.time_window,
+                "resolution": rule.resolution,
+                # TODO: Remove when frontend isn't using
+                "alertThreshold": 0,
+                # TODO: Remove when frontend isn't using
+                "resolveThreshold": 0,
+                "thresholdPeriod": rule.threshold_period,
+                "triggers": attrs.get("triggers", []),
+                "includeAllProjects": rule.include_all_projects,
+                "dateModified": rule.date_modified,
+                "dateAdded": rule.date_added,
+            }
+        elif isinstance(rule, Rule):
+            # serializer = AlertRuleSerializer()
+            environment = None  # attrs["environment"]
+            return {
+                "type": "rule",
+                "id": six.text_type(rule.id) if rule.id else None,
+                "conditions": [
+                    dict(o.items() + [("name", _generate_rule_label(rule.project, obj, o))])
+                    for o in rule.data.get("conditions", [])
+                ],
+                "actions": [
+                    dict(o.items() + [("name", _generate_rule_label(rule.project, obj, o))])
+                    for o in rule.data.get("actions", [])
+                ],
+                "actionMatch": rule.data.get("action_match") or Rule.DEFAULT_ACTION_MATCH,
+                "frequency": rule.data.get("frequency") or Rule.DEFAULT_FREQUENCY,
+                "name": rule.label,
+                "dateCreated": rule.date_added,
+                "environment": environment.name if environment is not None else None,
+            }
+        else:
+            raise AssertionError("Invalid rule to serialize: %r" % type(rule))
+
+
+class ProjectCombinedRuleIndexEndpoint(ProjectEndpoint):
+    def get(self, request, project):
+        """
+        Fetches alert rules and legacy rules for an organization
+        """
+        if not features.has("organizations:incidents", project.organization, actor=request.user):
+            raise ResourceDoesNotExist
+
+        cursor_string = request.GET.get("cursor", None)
+        page_size = 25
+
+        if cursor_string is None:
+            cursor_string = "0:0:0"
+
+        cursor = Cursor.from_string(cursor_string)
+        cursor_date = datetime.fromtimestamp(float(cursor.value)).replace(tzinfo=timezone.utc)
+
+        alert_rule_queryset = (
+            AlertRule.objects.fetch_for_project(project).order_by("-date_added")
```

Comment:
I usually do `order_by` after `filter`. I'm not sure if this changes how Django writes the SQL, could be worth checking